Java/core/src/main/java/com/amazon/randomcutforest/state/sampler/CompactSamplerMapper.java

```diff
@@ -69,7 +69,8 @@ public CompactSampler toModel(CompactSamplerState state, long seed) {
         }
 
         return new CompactSampler(state.getCapacity(), state.getSize(), state.getLambda(), new Random(seed), weight,
-                pointIndex, sequenceIndex, validateHeap);
+                pointIndex, sequenceIndex, validateHeap, state.getLargestSequenceIndexSeen(),
```

Comment:
What if, as part of the mapping, we adjusted the sampler weights by the accumulated lambda value? Then I think we would only need to serialize the maximum sequence index field. When a new sampler is created from the state, last update time is set to maximum sequence index and accumulated lambda is set to 0.